assets/src/phoenix/app.ts

```diff
@@ -35,8 +35,18 @@ const liveSocket = new LiveSocket('/live', Socket, {
 });
 
 // Show progress bar on live navigation and form submits
-window.addEventListener('phx:page-loading-start', (_info) => NProgress.start());
-window.addEventListener('phx:page-loading-stop', (_info) => NProgress.done());
+// this only shows the topbar if it's taking longer than 200 msec to receive the phx:page-loading-stop event
+let topBarScheduled: NodeJS.Timeout | undefined = undefined;
```

Comment:
NIT: If a variable is not initialized, it's going to be undefined by default so no need to assign it